pom.xml

```diff
@@ -939,7 +938,125 @@
+                                     <!-- disallow undocumented classes like sun.misc.Unsafe: -->
+                                     <internalRuntimeForbidden>true</internalRuntimeForbidden>
+                                     <!-- if the used Java version is too new, don't fail, just do nothing: -->
+                                     <failOnUnsupportedJava>false</failOnUnsupportedJava>
+                                     <!-- maven shade plugin is a nightmare -->
+                                     <failOnUnresolvableSignatures>false</failOnUnresolvableSignatures>
+                                     <excludes>
+                                         <exclude>jsr166e/**</exclude>
+                                     </excludes>
+                                     <bundledSignatures>
+                                         <!-- This will automatically choose the right signatures based on 'targetVersion': -->
+                                         <bundledSignature>jdk-unsafe</bundledSignature>
+                                         <bundledSignature>jdk-deprecated</bundledSignature>
+                                         <bundledSignature>jdk-system-out</bundledSignature>
+                                     </bundledSignatures>
+                                     <signaturesFiles>
+                                          <signaturesFile>${project.build.directory}/dev-tools/forbidden/core-signatures.txt</signaturesFile>
+                                          <signaturesFile>${project.build.directory}/dev-tools/forbidden/all-signatures.txt</signaturesFile>
+                                     </signaturesFiles>
+                                     <signatures>${forbidden.signatures}</signatures>
+                                     <suppressAnnotations><annotation>**.SuppressForbidden</annotation></suppressAnnotations>
+                                 </configuration>
+                             <phase>compile</phase>
+                             <goals>
+                                 <goal>check</goal>
+                             </goals>
+                         </execution>
+                         <execution>
+                             <id>check-forbidden-test-apis</id>
+                             <configuration>
+                                 <targetVersion>1.7</targetVersion>
+                                 <!-- disallow undocumented classes like sun.misc.Unsafe: -->
+                                 <internalRuntimeForbidden>true</internalRuntimeForbidden>
+                                 <!-- if the used Java version is too new, don't fail, just do nothing: -->
+                                 <failOnUnsupportedJava>false</failOnUnsupportedJava>
+                                 <!-- maven shade plugin is a nightmare -->
+                                 <failOnUnresolvableSignatures>false</failOnUnresolvableSignatures>
```

Comment:
Guys try to ignore this for now :)

Just to explain the issue: forbidden-apis fails by default if any of your signatures cannot be resolved in the classpath. This is a good thing, and I want that behavior back. This is a temporarily solution to keep changes minimal and keep in sync with elasticsearch-core at the moment.

The issue is that elasticsearch-core signatures are defined as unshaded class names, but plugins depend on shaded versions. 

One solution would be to fix es-core build to somehow run with shaded classpath. To me thats the real bug and the current inconsistency (that it tests with unshaded classpath but runs in production with a shaded one) scares me a lot. For example, we can't tighten our security policies at the package level because of this.

Another solution is to just pull out the 3rd party signatures into thirdparty.txt and have thirdparty_shaded.txt that goes with it.